Doppler.BigQueryMicroservice/Repository/Classes/BaseRepository.cs

```diff
@@ -1,29 +1,83 @@
+using Dapper;
 using Doppler.BigQueryMicroservice.Infrastructure;
+using Doppler.BigQueryMicroservice.Repository.Interfaces;
 using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
 using System.Data;
+using System.Linq;
 using System.Threading.Tasks;
 
 namespace Doppler.BigQueryMicroservice.Repository.Classes
 {
     /// <summary>
     /// Base repository abstraction
     /// </summary>
-    public abstract class BaseRepository
+    public abstract class BaseRepository<T> : IBaseRepository<T> where T : class
     {
-        private readonly ILogger<BaseRepository> _logger;
+        private readonly ILogger<T> _logger;
         private readonly IDatabaseConnectionFactory _connectionFactory;
-        public BaseRepository(IDatabaseConnectionFactory connectionFactory, ILogger<BaseRepository> bigQueryLogger)
+        public string TableName { get; set; }
+
+        public BaseRepository(IDatabaseConnectionFactory connectionFactory, ILogger<T> bigQueryLogger)
         {
             _connectionFactory = connectionFactory;
             _logger = bigQueryLogger;
         }
 
-
         protected async Task<IDbConnection> CreateConnectionAsync()
         {
             return await _connectionFactory.GetConnection();
         }
 
-        public ILogger<BaseRepository> BigQueryLogger { get { return _logger; } }
+        public ILogger<T> BigQueryLogger { get { return _logger; } }
+
+        #region BaseRepository implementation
+        public async Task<T> GetByIdAsync(int id)
+        {
+            var builder = new SqlBuilder();
+            builder.Select("*").Where($"Id = @Id");
+            var builderTemplate = builder.AddTemplate($"Select /**select**/ from {TableName} /**where**/");
+
+            using (var connection = await CreateConnectionAsync())
+            {
+                try
+                {
+                    var result = await connection.QuerySingleOrDefaultAsync<T>(builderTemplate.RawSql, new { Id = id });
+                    return result;
+                }
+                catch (Exception ex)
+                {
+                    BigQueryLogger.LogError(ex, $"Entity {nameof(T)} by id not exist id:{id}");
+                    throw;
+                }
+                finally
+                {
+                    connection.Close();
+                }
+            }
+        }
+
+        public Task<System.Collections.Generic.IReadOnlyList<T>> GetAllAsync()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<int> AddAsync(T entity)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<int> UpdateAsync(T entity)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public Task<int> DeleteAsync(int id)
+        {
+            throw new System.NotImplementedException();
+        }
```

Comment:
Creo que no necesitamos repositorios genéricos. Si vamos a re-usar me parece que podemos utilizar composición en lugar de herencia y listo.

De esa forma evitaríamos tener este tipo de problemas.

Reply:
se eliminan los metodos no implementados del repositorio

Reply:
en el proximo PR refactor de las cosas innecesarias como el unit of work